src/util-ioctl.c

```diff
@@ -124,21 +114,26 @@ int GetIfaceMTU(const char *pcap_dev)
  * for the link. In case of uncertainty, it will output a
  * majorant to be sure avoid the cost of dynamic allocation.
  *
- * \param Name of a network interface
+ * \param LiveDevice object
  * \retval 0 in case of error
  */
-int GetIfaceMaxPacketSize(const char *pcap_dev)
+int GetIfaceMaxPacketSize(LiveDevice *ld)
 {
-    if ((pcap_dev == NULL) || strlen(pcap_dev) == 0)
+    if (ld == NULL)
+        return 0;
+
+    const char *dev = ld->dev;
+    if ((dev == NULL) || strlen(dev) == 0)
         return 0;
 
-    int mtu = GetIfaceMTU(pcap_dev);
+    int mtu = GetIfaceMTU(dev);
     switch (mtu) {
         case 0:
         case -1:
             return 0;
     }
-    int ll_header = GetIfaceMaxHWHeaderLength(pcap_dev);
+    ld->mtu = mtu;
+    int ll_header = GetIfaceMaxHWHeaderLength(dev);
```

Comment:
`GetIfaceMaxHWHeaderLength` will never return -1. Even though this is a pre-existing issue, I'd suggest removing the check against `-1`

Reply:
Done. Please check the latest PR.